avocado/utils/genio.py

```diff
@@ -174,3 +174,16 @@ def write_one_line(filename, line):
     :type line: str
     """
     write_file(filename, line.rstrip('\n') + '\n')
+
+
+class GenIOException(Exception):
+    pass
+
+
+def write_file_or_log(filename, data):
```

Comment:
Oh and would you please also describe what this does? (I mean the difference between this and the other classes)